### DISCUSSION TITLE
Stumped by an initial very simple grammar

### DISCUSSION THREAD

**Discussion by The-Futurist** [antlr/antlr4#4012] [Q&A]
I'm reading through Terence Parr's book and experimenting as I go, I'm a Windows user.

The tools seem to work, everything it setup and I'm generating Java for now (will do C# later when I'm more accustomed to all this).

Anyway, I'm stumped, a tiny grammar (literally a tweak from what's in the book) causes me issues, look, here's the grammar and a sample input file:

![image](https://user-images.githubusercontent.com/12262952/207656200-f860d6d2-f8f1-46b7-b34b-1707e04b1e30.png)

This is the output from the `grun` command:

![image](https://user-images.githubusercontent.com/12262952/207656480-725fc1f4-1257-46a5-a761-7738afddaf31.png)

I cannot understand why the simplest input `a=1` causes a problem! Surely a single digit `1` meets the definition of `INT` in that grammar?

Here is the file itself as a raw data dump:

![image](https://user-images.githubusercontent.com/12262952/207657338-51aea50f-108f-4bbf-8c91-8ead7aceeb53.png)

If I change the definition of `INT` to be simply `INT: '1' ;` then it parses that file fine.

## Resolved

Ok its Unicode again, here is what's actually in the `g4` file:

![image](https://user-images.githubusercontent.com/12262952/207666015-579103f5-0670-4b63-b28d-459ed42bc2c9.png)

Note, that's actually a 'EN DASH' character with code `E28093` - see [here](https://www.fileformat.info/info/unicode/char/2013/index.htm).

**Comment by kaby76** [comment]
Some suggestions:
* Pitch the Java target and just use the C# target using Antlr4BuildTasks with Antlr4.Runtime.Standard. Here is a sample project set up for you.
[Generated-CSharp.zip](https://github.com/antlr/antlr4/files/10231775/Generated-CSharp.zip)
 After unpacking, type "dotnet run whatever-your-file-is" and the program will parse the input file.
* If you need a tree, go to lab.antlr.org and paste in your grammar and input. The output tree can be saved as .svg.
* Use the "-tokens" option to output a list of tokens so you know what the heck is going on with a parse. The prereq for a parse is a lex--make sure the tokens are right.
* Set your editor to use "utf8 no bom" or just encode the files as ASCII.
